unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1965,18 +1962,42 @@ protected void updateAllAgentCollidersForVisibilityCheck(bool enableColliders)
             return Time.time;
         }
 
-        private bool isSimObjVisible(Camera agentCamera, SimObjPhysics sop) {
+        private bool isSimObjVisible(Camera agentCamera, SimObjPhysics sop, float maxDistance) {
             bool visible = false;
             //check against all visibility points, accumulate count. If at least one point is visible, set object to visible
-            if (sop.VisibilityPoints == null || sop.VisibilityPoints.Length > 0) 
+            if (sop.VisibilityPoints != null && sop.VisibilityPoints.Length > 0) 
             {
                 Transform[] visPoints = sop.VisibilityPoints;
                 int visPointCount = 0;
 
                 foreach (Transform point in visPoints) 
                 {
+                    float xdelta = Math.Abs(this.transform.position.x - point.position.x);
+                    if (xdelta > maxDistance) 
+                    {
+                        continue;
+                    }
+
+                    float zdelta = Math.Abs(this.transform.position.z - point.position.z);
+                    if (zdelta > maxDistance) 
+                    {
+                        continue;
+                    }
```

Comment:
Why are these if statements necessary? Since `abs(xdelta) <= distance` and `abs(zdelta) <= distance` aren't these calls effectively redundant in view of the `if (distance > maxDistance) {` statement below?

Reply:
The goal is to short circuit as early as possible and avoid having to compute the distance.  Let me benchmark without them though and see if it is actually contributing.

Reply:
I just benchmarked this and keeping the shortcircuits in results in ~20% performance improvement.